src/IW/Sync/Search.hs

```diff
@@ -0,0 +1,89 @@
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchHaskellIssuesByLabels
+       , githubIssueToIssue
+       , githubRepoToRepo
+       ) where
+
+import Data.Text (split)
+import GitHub (SearchResult (..))        
+import GitHub.Endpoints.Search (searchRepos, searchIssues)
+
+import IW.App (WithError)
+import IW.Core.Issue (Issue (..), Label (..))
+import IW.Core.Repo (Repo (..), RepoOwner (..), RepoName (..))
+import IW.Core.SqlArray (SqlArray (..))
+import IW.Db (WithDb)
+import IW.App.Error (githubErrToAppErr, throwError)
+
+import qualified GitHub
+import qualified Data.Vector as Vector
+
+
+-- | Fetch all repositories built with the Haskell language.
+fetchAllHaskellRepos :: (WithDb env m, WithError m, WithLog env m) => m [GitHub.Repo]
+fetchAllHaskellRepos = liftGitHubSearchToApp searchHaskellRepos 
+
+-- | Fetch all open issues with Haskell language and the labels passed in to the function.
+fetchHaskellIssuesByLabels :: (WithDb env m, WithError m, WithLog env m) => [Label] -> m [GitHub.Issue]
+fetchHaskellIssuesByLabels labels = liftGitHubSearchToApp $ searchHaskellIssuesByLabels labels
+
+-- | Fetch all open issues with Haskell language.
+fetchAllHaskellIssues :: (WithDb env m, WithError m, WithLog env m) => m [GitHub.Issue]
+fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
+
+-- | Lift a github search action to work within our monad stack.
+liftGitHubSearchToApp :: (WithDb env m, WithError m, WithLog env m) => IO (Either GitHub.Error (SearchResult a)) -> m [a]
+liftGitHubSearchToApp githubSearch = do
+    result <- liftIO githubSearch
```

Comment:
This is actually a nice use case for `LambdaCase` extension. You can write:

```haskell
liftGitHubSearchToApp searchAction = liftIO searchAction >>= \case
    Left ...
    Right ...
```

This helps to avoid need to come up with the variable names and make nested indentation less nested.